server/src/main/java/io/crate/execution/ddl/tables/MappingUtil.java

```diff
@@ -185,11 +202,20 @@ private static Map<String, Map<String, Object>> createPropertiesMap(AllocPositio
         }
         HashMap<String, Map<String, Object>> allColumnsMap = new LinkedHashMap<>();
         for (Reference child: children) {
-            allColumnsMap.put(child.column().leafName(), addColumnProperties(position, child, tree));
+            allColumnsMap.put(mappingKey(child), addColumnProperties(position, child, tree));
         }
         return allColumnsMap;
     }
 
+    private static String mappingKey(Reference reference) {
+        if (reference.isDropped()) {
+            assert reference.oid() != COLUMN_OID_UNASSIGNED : "Only columns with assigned OID-s can be dropped";
```

Comment:
need Flag check since parent caller function is used in `AddColumnTask` where we need normal column names.